manifests.go

```diff
@@ -80,7 +81,17 @@ var mappings = make(map[string]UnmarshalFunc, 0)
 
 // UnmarshalManifest looks up manifest unmarshall functions based on
 // MediaType
-func UnmarshalManifest(mediatype string, p []byte) (Manifest, Descriptor, error) {
+func UnmarshalManifest(ctHeader string, p []byte) (Manifest, Descriptor, error) {
+	// Need to look up by the actual content type, not the raw contents of
+	// the header. Strip semicolons and anything following them.
+	var mediatype string
+	semicolonIndex := strings.Index(ctHeader, ";")
```

Comment:
@aaronlehmann That's up to you. I don't know the header format well enough to understand the edge cases, so I am not sure if we will have any issues here.